src/Microsoft.AspNet.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -224,18 +228,24 @@ public void AddTagHelperAttribute(string name, object value)
         }
 
         /// <summary>
-        /// Execute and retrieve the rendered child content asynchronously.
+        /// Execute children asynchronously with the given <paramref name="encoder"/> in scope and return their
+        /// rendered content.
         /// </summary>
+        /// <param name="useCachedResult">If <c>true</c> multiple calls to this method will not cause re-execution
+        /// of child content; cached content will be returned.</param>
+        /// <param name="encoder">
+        /// The <see cref="HtmlEncoder"/> to use. Does not override all HTML encodings which may occur.
+        /// </param>
         /// <returns>A <see cref="Task"/> that on completion returns the rendered child content.</returns>
         /// <remarks>
         /// Child content is only executed once. Successive calls to this method or successive executions of the
         /// returned <see cref="Task{TagHelperContent}"/> return a cached result.
         /// </remarks>
-        public async Task<TagHelperContent> GetChildContentAsync(bool useCachedResult)
+        public async Task<TagHelperContent> GetChildContentAsync(bool useCachedResult, HtmlEncoder encoder)
```

Comment:
Forgot the 3rd approach ([mentioned](https://github.com/aspnet/Razor/issues/643#issuecomment-164317532) in #643).

##### Recommend `@Html.Raw()`
E.g. `<markdown>@Html.Raw(release.Body)</markdown>` or (as the original poster did) `<markdown>@(new HtmlString(release.Body))</markdown>`. This puts control in the `.cshtml` author's hands and tasks tag helper authors mainly with writing good doc comments for IntelliSense.

One upside is this maintains consistency w.r.t. to what is HTML encoded i.e. everything that isn't either in the `.cshtml` file itself or specially marked is encoded. The downside here is mainly discoverability, especially if the tag helper author does not write good comments or IntelliSense isn't working.